SonarQube.Client/Api/DefaultConfiguration.cs

```diff
@@ -54,7 +54,8 @@ public static RequestFactory ConfigureSonarQube(RequestFactory requestFactory)
                 .RegisterRequest<IGetIssuesRequest, V7_20.GetIssuesRequestWrapper>("7.2")
                 .RegisterRequest<IGetHotspotRequest, V8_6.GetHotspotRequest>("8.6")
                 .RegisterRequest<IGetTaintVulnerabilitiesRequest, V8_6.GetTaintVulnerabilitiesRequest>("8.6")
-                .RegisterRequest<IGetExclusionsRequest, V7_20.GetExclusionsRequest>("7.2");
+                .RegisterRequest<IGetExclusionsRequest, V7_20.GetExclusionsRequest>("7.2")
+                .RegisterRequest<IPushRequest, V9_4.PushRequest>("9.4");
```

Comment:
How are we going to handle the new API method call being made against SonarCloud?
Currently, I think the SC request factory will throw.

Is it the responsibility of the calling client to not call the new method if they are connected to SonarCloud, or the responsibility of this webclient layer to handle it (e.g. by returning a no-op response of some kind e.g. an empty/null stream)?

I think it would be preferable if the callers didn't need to know about differences between the SonarQube and SonarCloud.

Reply:
It would be the same as an unsupported old SQ version: the API will throw InvalidOperationException. 
It will then be up to the caller to know that we are not connected to a supported instance. This is what I meant in this comment https://github.com/SonarSource/sonarqube-webclient-dotnet/pull/142#discussion_r1067902116: 
`This is a bit ugly, the client will have to handle unsupported SQ versions and we return "InvalidOperationException". NotImplemented or UnsupportedMethod would've been more suitable IMO, but that's InvalidOperationException is what the infra currently returns.`

It would've been much better to return something else, i.e. NotSupportedException, but we can't change it without changing the infra of all the requests.

Reply:
I don't think the type of the exception that is thrown is necessarily the problem; we're still throwing something the client needs to handle.

I was suggesting that we could add a dummy request to the SonarCloud factory so the request doesn't fail, but instead returns a no-op e.g. a [null-object](https://en.wikipedia.org/wiki/Null_object_pattern). That way the caller wouldn't need to know about differences between SC and SQ.

It would mean the client would create instances of classes that are waiting for events that will never come, but the overhead of that is minimal.

Reply:
as we discussed offline, for now I'll leave it to SLVS